deidentification/anonymizer.py

```diff
@@ -16,83 +14,150 @@
+        
+        if anonymous:
+            tags_to_keep = load_config_anonymous(config_profile)
+        else:
+            tags_to_keep = load_config_profile(config_profile)
+    
     if not os.path.exists(dicom_folder_out):
         os.makedirs(dicom_folder_out)
     dicom_file_out = os.path.join(dicom_folder_out,
                                   os.path.basename(dicom_file_in))
 
     anon = Anonymizer(dicom_file_in, dicom_file_out,
-                      tags_to_keep, forced_values)
-    if not anon.run_ano():
-        print("File is a DICOMDIR.")
+                      tags_to_keep, forced_values,
+                      anonymous=anonymous)
+    anon.run_ano()
+
+
+def load_config_anonymous(config_profile):
+    load_error = ''
+    try:
+        tags_to_keep = load_config_profile(config_profile)
+    except (ValueError, AttributeError):
+        load_error = 'Error occurs during config profile load.'
+    if load_error:
+        raise DeidentificationError(load_error)
+    return tags_to_keep
 
 
 def anonymize(dicom_in, dicom_out,
-              tags_to_keep=tags_to_keep,
-              forced_values=forced_values):
-    """
-    Configures the Anonymizer and runs it on DICOM files.
+              tags_to_keep=None,
+              forced_values=None,
+              config_profile=None,
+              anonymous=False):
+    """Configures the Anonymizer and runs it on DICOM files.
+    
+    Parameters
+    ----------
+    dicom_in : str
+    dicom_out : str
+    tags_to_keep : list, optional
+    forced_values : dict, optional
+    config_profile : str, optional
+    anonymous : bool, optional
     """
     if not os.path.exists(dicom_in):
-        print("The DICOM input does not exist.")
-        return
+        raise DeidentificationError('The DICOM input does not exists.')
+    if not os.path.isfile(dicom_in) and not os.path.isdir(dicom_in):
+        raise DeidentificationError('The DICOM input file type is not handled by this tool.')
+    if not os.path.isfile(dicom_in) and os.path.isfile(dicom_out):
+        raise DeidentificationError('The DICOM out could not be a file if DICOM in is not a file.')
+    if config_profile:
+        if tags_to_keep:
+            raise DeidentificationError('Both tags_to_keep and config_profile have been specified.')
+        tags_to_keep = load_config_profile(config_profile)
 
+    # Handle archives
     is_dicom_in_archive = is_archive(dicom_in)
     is_dicom_out_archive = is_archive(dicom_out)
     if is_dicom_in_archive:
         wip_dicom_in = mkdtemp()
-        unpack(dicom_in, wip_dicom_in)
+        try:
+            unpack(dicom_in, wip_dicom_in)
+        except Exception:
+            raise DeidentificationError('Unpacking compressed file failed.')
     else:
         wip_dicom_in = os.path.abspath(dicom_in)
     if is_dicom_out_archive:
         wip_dicom_out = mkdtemp()
     else:
         wip_dicom_out = os.path.abspath(dicom_out)
```

Comment:
@Hboni , I think we should handle deletion of `wip_dicom_in` and `wip_dicom_out` in case of `unpack` failed. If `unpack` failed while it start, it is possible that we have `DeidentificationError`  and unpacked files remains.

Reply:
You are right, I modified this in anonymize and in check functions to avoid this case :+1: